packages/sdk/contractkit/src/wrappers/SortedOracles.test.ts

```diff
@@ -35,252 +40,432 @@ testWithGanache('SortedOracles Wrapper', (web3) => {
     }
 
     for (let i = 0; i < rates.length; i++) {
-      const tx = await sortedOracles.report(CeloContract.StableToken, rates[i], oracles[i])
+      const tx = await sortedOracles.report(target, rates[i], oracles[i])
       await tx.sendAndWaitForReceipt()
     }
   }
 
   // Quick setup for conditions when some oracle reports are expired and the rest are not.
   // This assumes that the rates reported can be arbitrary and not a critical piece of the test.
-  async function setupExpiredAndNotExpiredReports(expiredOracles: Address[]): Promise<void> {
+  async function setupExpiredAndNotExpiredReports(
+    sortedOracles: SortedOraclesWrapper,
+    target: ReportTarget,
+    expiredOracles: Address[],
+    allOracles: Address[]
+  ): Promise<void> {
     const expirySeconds = (await sortedOracles.reportExpirySeconds()).toNumber()
-    await reportAsOracles(expiredOracles)
+    await reportAsOracles(sortedOracles, target, expiredOracles)
     await timeTravel(expirySeconds + 5, web3)
-    const freshOracles = stableTokenOracles.filter((o) => !expiredOracles.includes(o))
-    await reportAsOracles(freshOracles)
+    const freshOracles = allOracles.filter((o) => !expiredOracles.includes(o))
+    await reportAsOracles(sortedOracles, target, freshOracles)
   }
 
-  beforeAll(async () => {
-    sortedOracles = await kit.contracts.getSortedOracles()
-    stableTokenAddress = await kit.registry.addressFor(CeloContract.StableToken)
-    allAccounts = await web3.eth.getAccounts()
-    nonOracleAddress = allAccounts.find((addr) => {
-      return !stableTokenOracles.includes(addr)
-    })!
-  })
+  /**
+   * When testing with a custom token pair we can't use the
+   * already deployed SortedOracles because that's managed
+   * by Governance and we can't execute changes on the contract.
+   * To make it easier we'll deploy an new version for use in
+   * the tests
+   */
+  async function newSortedOracles(owner: Address): Promise<SortedOraclesWrapper> {
+    const instance = await SortedOracles.new({ from: owner })
+    await instance.initialize(NetworkConfig.oracles.reportExpiry, { from: owner })
+    return new SortedOraclesWrapper(kit, instance.contract)
+  }
 
-  describe('#report', () => {
-    const value = 16
+  async function addOracleForPair(
+    sortedOraclesInstance: SortedOraclesWrapper,
+    pair: PairWithIdentifier,
+    oracle: Address,
+    owner: Address
+  ): Promise<void> {
+    // @ts-ignore
```

Comment:
just curious, does `SortedOraclesWrapper` not expose the contract in a TS-friendly way?

Reply:
Yeah it's a private property, it makes sense in that the wrapper should be the outside interface to the contract and theoretically the wrapper should expose methods that wrap the contract methods, but in this particular case oracles can only be added through governance in usual circumstances so probably that's why the wrapper doesn't implement that function.

Reply:
Can you add a short version of that explanation after `// @ts-ignore`?